Test/civl/inductive-sequentialization/PingPong.bpl

```diff
@@ -1,16 +1,32 @@
 // RUN: %parallel-boogie "%s" > "%t"
 // RUN: %diff "%s.expect" "%t"
 
-type {:linear "pid"} Pid = int;
-var {:layer 0,3} ping_channel:[int]int; // Incoming channel of Ping
-var {:layer 0,3} pong_channel:[int]int; // Incoming channel of Pong
+// This example shows how to use a bidirectional shared channel to communicate between two processes.
+
+// A bidirectional channel is a pair of ordinary channels with two ends---left and right.
+type {:datatype} ChannelPair;
+function {:constructor} ChannelPair(left: [int]int, right: [int]int): ChannelPair;
+
+type ChannelId;
+var {:layer 0,3} channel: [ChannelId]ChannelPair;
+
+// The id of a bidirectional channel can be split into two permissions---Left and Right.
+// Left permission is used to receive from the left channel and send to the right channel.
+// Right permission is used to receive from the right channel and send to the left channel.
+type {:linear "cid"} {:datatype} Permission;
+function {:constructor} Left(cid: ChannelId): Permission;
+function {:constructor} Right(cid: ChannelId): Permission;
+function {:inline} ChannelId(p: Permission) : ChannelId {
+  if is#Left(p) then cid#Left(p) else cid#Right(p)
+}
```

Comment:
I'm wondering if renaming `Permission` to something like `ChannelHandle` makes sense.

In any case, I think there should be an explanation here or at the definitions of `SEND`/`RECEIVE` (or both) that the name of a permission/handle comes from the receiving end. I.e., with a `Left` we receive from the left channel and send to the right, and with a `Right` we recieve from the right channel and send to the left.

Reply:
I have renamed Permission to ChannelHandle.  

I didn't understand what you meant by the extra comment. The existing comment seems to be doing what you want.

Reply:
> I didn't understand what you meant by the extra comment. The existing comment seems to be doing what you want.

Oh right, sorry.